lib/sidekiq/throttled/strategy/concurrency.rb

```diff
@@ -76,7 +82,44 @@ def reset!(*job_args)
         # Remove jid from the pool of jobs in progress
         # @return [void]
         def finalize!(jid, *job_args)
-          Sidekiq.redis { |conn| conn.zrem(key(job_args), jid.to_s) }
+          Sidekiq.redis do |conn|
+            conn.zrem(key(job_args), jid.to_s)
+          end
+        end
+
+        private
+
+        def backlog_info_key(job_args)
+          "#{key(job_args)}.backlog_info"
+        end
+
+        def estimated_backlog_size(job_args)
+          old_size_str, old_timestamp_str =
+            Sidekiq.redis { |conn| conn.hmget(backlog_info_key(job_args), "size", "timestamp") }
+          old_size = (old_size_str || 0).to_f
+          old_timestamp = (old_timestamp_str || Time.now).to_f
+
+          nonneg(old_size - jobs_lost_since(old_timestamp, job_args))
```

Comment:
IMO it's better to use `clamp` than add new method:
```suggestion
          (old_size - jobs_lost_since(old_timestamp, job_args)).clamp(0, nil)
```

Reply:
good catch